### ISSUE TITLE
Consolidate quickstart ownerReference tests

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes-sigs/cluster-api#7961]
#7575 adds two tests for ownerReference resilience. These tests, once they're stable, should be added to the ordinary quickstart and ClusterClass quickstart tests as an additional assertion in order to reduce the overall number of tests running.

/kind feature

**Comment by fabriziopandini** [comment]
Might be we should use some other tests we can use for this scope because IMO we should keep the quick start test as fast as possible because we run them on every PR.

Side note, I was thinking about the overall number of tests we are running from a different angle, and starting to consider if/how to split our E2E full job in two jobs, might be one for cluster classes and one not...

**Comment by sbueringer** [comment]
If I saw correctly the tests with ownerref validation are still very fast (~ 2 m when they are run as part of e2e-full, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/7606/pull-cluster-api-e2e-full-main/1616453975815491584)

**Comment by killianmuldoon** [comment]
Yeah - it's very quick, but I agree that we might be able to take the opportunity to de-deupe a number of tests at the same time.

**Comment by killianmuldoon** [comment]
Just to follow up on this. This test has been 100 percent green since being introduced on main ([and green since an implementation issue was fixed for the minK8s run](https://github.com/kubernetes-sigs/cluster-api/pull/7973)). 

Time-wise this test has a similar runtime and variance to the quick-start tests. I think logically it will make those tests longer, but practically the difference is in the magnitude of seconds, and is dwarfed by the normal variance in the Cluster creation workflow.

Definitely think we should leave the tests to run for another week or two for a stronger signal of flakiness, but I don't think runtime should be a deciding factor in whether or not to include it as part of the PR Blocking / Informing tests.

**Comment by sbueringer** [comment]
Sounds fine to me!
